ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/StateTestSubCommand.java

```diff
@@ -248,14 +251,31 @@ private void traceTestSpecs(final String test, final List<GeneralStateTestCaseEi
         final Stopwatch timer = Stopwatch.createStarted();
         // Todo: EIP-4844 use the excessBlobGas of the parent instead of BlobGas.ZERO
         final Wei blobGasPrice = protocolSpec.getFeeMarket().blobGasPricePerGas(BlobGas.ZERO);
+
+        BlockHashOperation.BlockHashLookup blockHashLookup =
+            protocolSpec
+                .getBlockHashProcessor()
+                .getBlockHashLookup(
+                    blockHeader, new ReferenceTestBlockchain(blockHeader.getNumber()));
+        if (blockHashLookup instanceof CachingBlockHashLookup) {
+          blockHashLookup =
+              (messageFrame, number) -> {
+                long lookback = messageFrame.getBlockValues().getNumber() - number;
+                if (lookback < 0 || lookback > BlockHashOperation.MAX_RELATIVE_BLOCK) {
```

Comment:
SHould be `lookback <= 0`, probably. Because that means the messageframe blocknumber is the same as the one requested: `BLOCKHASH(this block)`.